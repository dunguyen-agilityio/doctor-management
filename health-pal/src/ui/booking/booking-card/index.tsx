import { TBookingCard } from '@/models/booking'
import { formatDate } from '@/utils/date'

import { Card, Separator } from 'tamagui'

import { Heading, Text, XStack, YStack } from '@/components'
import CloudinaryImage from '@/components/cloudinary-image'

import { BOOKING_TABS } from '@/types/booking'

import BookingAction from './booking-action'

const BookingCard = ({
  date,
  time,
  doctorName,
  specialty,
  doctorAvatar,
  address,
  documentId,
  doctorId,
  doctorDocId,
  ...props
}: TBookingCard) => {
  const { type = BOOKING_TABS.UPCOMING } = props

  return (
    <Card
      borderRadius={12}
      borderWidth={0.5}
      borderColor="$grey100"
      backgroundColor="$white"
      shadowColor="$black"
      shadowOffset={{ width: 0, height: 4 }}
      shadowOpacity={0.1}
      shadowRadius={6}
      elevation={3}
      padding={10}
      marginBottom={10}
      testID="booking-card"
      aria-label={`Booking with ${doctorName}`}>
      <Card.Header>
        <Heading fontSize={14}>{formatDate(date)}</Heading>
        <Separator marginVertical={12} />
      </Card.Header>

      <Card.Footer>
        <YStack flex={1}>
          <XStack gap={10}>
            <CloudinaryImage
              source={{ uri: doctorAvatar }}
              style={{ height: 109, width: 109, borderRadius: 8 }}
            />
            <YStack paddingVertical={14}>
              <Text fontSize={16} fontWeight="bold">
                {doctorName}
              </Text>
              <Text fontSize={14} color="$grey500">
                {specialty}
              </Text>
              <Text fontSize={12} color="$grey400" numberOfLines={1} maxWidth={250}>
                {address}
              </Text>
            </YStack>
          </XStack>
          <BookingAction
            date={date}
            doctorDocId={doctorDocId}
            doctorId={doctorId}
            doctorName={doctorName}
            documentId={documentId}
            type={type}
          />
        </YStack>
      </Card.Footer>
    </Card>
  )
}

export default BookingCard
