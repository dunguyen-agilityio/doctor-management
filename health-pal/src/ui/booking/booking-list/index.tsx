import { FlashList } from '@shopify/flash-list'

import { StyleSheet } from 'react-native'

import { useQuery } from '@tanstack/react-query'

import { BOOKING_EMPTY } from '@app/constants'

import { Empty, ErrorState } from '@app/components'
import BookingListSkeleton from '@app/components/skeleton/booking-list-skeleton'

import { getBookings } from '@app/services/booking'

import { BOOKING_TABS } from '@app/types/booking'
import { StrapiPagination } from '@app/types/strapi'

import { BookingData, BookingKey } from '@app/models/booking'
import { formatBooking } from '@app/utils/booking'
import { keyExtractor } from '@app/utils/list'
import { getMediaQuery } from '@app/utils/media-query'

import BookingCard from '../booking-card'

const bookingPromises: Record<BOOKING_TABS, () => Promise<StrapiPagination<BookingData>>> = {
  [BOOKING_TABS.CANCELED]: () =>
    getBookings({ filters: [{ key: BookingKey.type, query: BOOKING_TABS.CANCELED }] }),
  [BOOKING_TABS.COMPLETED]: () =>
    getBookings({ filters: [{ key: BookingKey.type, query: BOOKING_TABS.COMPLETED }] }),
  [BOOKING_TABS.UPCOMING]: () =>
    getBookings({ filters: [{ key: BookingKey.type, query: BOOKING_TABS.UPCOMING }] }),
}

const BookingList = ({ type }: { type: BOOKING_TABS }) => {
  const { width } = getMediaQuery({ full: true, width: 342, px: 24 })

  const { isLoading, isFetching, data, error, refetch } = useQuery({
    queryKey: ['bookings', type],
    queryFn: bookingPromises[type],
  })

  const renderItem = ({ item }: { item: BookingData }) => <BookingCard {...formatBooking(item)} />

  if (isLoading || isFetching) {
    return <BookingListSkeleton type={type} />
  }

  if (error || !data) {
    return (
      <ErrorState
        message={error?.message ?? 'Failed to fetch!'}
        aria-label="Error loading bookings"
        accessibilityHint="Failed to load bookings, press retry to try again"
        onRetry={refetch}
      />
    )
  }

  const ListEmptyComponent = (
    <Empty
      {...BOOKING_EMPTY[type]}
      aria-label={`No ${type.toLowerCase()} bookings`}
      accessibilityHint={`No ${type.toLowerCase()} bookings available`}
    />
  )

  return (
    <FlashList
      renderItem={renderItem}
      keyExtractor={keyExtractor}
      data={data.data}
      showsHorizontalScrollIndicator={false}
      showsVerticalScrollIndicator={false}
      contentContainerStyle={styles.contentContainerStyle}
      ListEmptyComponent={ListEmptyComponent}
      estimatedItemSize={width}
      aria-label={`${type} bookings list`}
      accessibilityHint={`List of ${type.toLowerCase()} bookings`}
      role="list"
    />
  )
}

export default BookingList

const styles = StyleSheet.create({
  contentContainerStyle: { paddingHorizontal: 24 },
})
