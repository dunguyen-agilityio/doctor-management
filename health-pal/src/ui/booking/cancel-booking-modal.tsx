import { useImperativeHandle, useRef, useState } from 'react'

import { router } from 'expo-router'

import { CheckCircle } from '@tamagui/lucide-icons'

import { Button, Heading, Text, YStack } from '@theme'

import { Modal } from '@app/components'
import { useSession } from '@app/contexts'
import { useAppLoading } from '@app/hooks'
import { TBookingCard } from '@app/models/booking'
import { queryClient } from '@app/react-query.config'
import { updateBooking } from '@app/services/booking'
import { BOOKING_TABS } from '@app/types/booking'
import { ModalRef } from '@app/types/modal'

export type CancelBookingParams = Pick<TBookingCard, 'date' | 'documentId' | 'time' | 'doctorName'>

type Props = {
  ref?: React.Ref<ModalRef<CancelBookingParams>>
}

const CancelBookingModal = ({ ref }: Props) => {
  const cancelBookingModalRef = useRef<ModalRef>(null)

  const [booking, setBooking] = useState<CancelBookingParams | null>(null)
  const { session } = useSession()
  const jwt = session?.jwt
  const setAppLoading = useAppLoading()

  const handleClose = () => {
    setBooking(null)
    cancelBookingModalRef.current?.close()
  }

  useImperativeHandle(ref, () => ({
    open: (params?: CancelBookingParams) => {
      if (params) {
        setBooking(params)
        cancelBookingModalRef.current?.open()
      }
    },
    close: handleClose,
  }))

  const handleCancelBooking = async () => {
    if (!jwt || booking === null) return

    try {
      setAppLoading(true)
      const { documentId } = booking

      handleClose()

      const response = await updateBooking({ documentId, type: BOOKING_TABS.CANCELED }, jwt)

      if (response.data) {
        await queryClient.invalidateQueries({ queryKey: [`bookings-${BOOKING_TABS.CANCELED}`] })
        await queryClient.invalidateQueries({ queryKey: [`bookings-${BOOKING_TABS.UPCOMING}`] })
        router.navigate({
          pathname: '/(app)/(tabs)/bookings',
          params: { type: BOOKING_TABS.CANCELED },
        })
      }
    } catch (error) {
      console.log('error', error)
    }

    setAppLoading(false)
  }

  return (
    <Modal ref={cancelBookingModalRef} closeButtonShown>
      <YStack alignItems="center" paddingHorizontal={42} gap={32} paddingVertical={32}>
        <YStack
          maxHeight={100}
          width={100}
          borderRadius={100}
          backgroundColor="#e0f7f9"
          alignItems="center"
          justifyContent="center">
          <CheckCircle color="$teal" size={50} />
        </YStack>

        <Heading size="extraLarge">Confirm Cancellation</Heading>

        <Text size="small" textAlign="center">
          Are you sure you want to cancel your appointment with{' '}
          <Text size="small" fontWeight="700">
            {booking?.doctorName}
          </Text>{' '}
          on{' '}
          <Text size="small" fontWeight="700">
            {booking?.date}
          </Text>
          ?
        </Text>

        <YStack gap={10} width="100%">
          <Button onPress={handleCancelBooking}>Yes, Cancel</Button>
          <Button
            variant="secondary"
            backgroundColor="transparent"
            color="$teal"
            borderWidth={0}
            onPress={handleClose}>
            No, Keep Appointment
          </Button>
        </YStack>
      </YStack>
    </Modal>
  )
}

export default CancelBookingModal
