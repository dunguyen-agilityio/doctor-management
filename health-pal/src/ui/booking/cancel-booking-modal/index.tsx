import { router } from 'expo-router'

import { useToastController } from '@tamagui/toast'

import { useAppLoading } from '@app/hooks'

import { Button, Heading, Modal, Text, YStack } from '@app/components'
import { ModalProps } from '@app/components/common/modal'

import { updateBooking } from '@app/services/booking'

import { CheckCircle } from '@icons'

import { BOOKING_TABS } from '@app/types/booking'
import { ModalRef } from '@app/types/modal'

import { TBookingCard } from '@app/models/booking'
import { queryClient } from '@app/react-query.config'
import { formatDate } from '@app/utils/date'

export type CancelBookingParams = Pick<TBookingCard, 'date' | 'documentId' | 'doctorName'>

interface Props extends CancelBookingParams, ModalProps {
  ref: React.RefObject<ModalRef | null>
}

const CancelBookingModal = ({ ref, date, doctorName, documentId, ...props }: Props) => {
  const setAppLoading = useAppLoading()
  const toast = useToastController()

  const handleClose = () => {
    ref.current?.close()
  }

  const handleCancelBooking = async () => {
    try {
      setAppLoading(true)

      handleClose()

      const response = await updateBooking({ documentId, type: BOOKING_TABS.CANCELED })

      if (response.data) {
        await queryClient.invalidateQueries({ queryKey: ['bookings', BOOKING_TABS.CANCELED] })
        await queryClient.invalidateQueries({ queryKey: ['bookings', BOOKING_TABS.UPCOMING] })
        router.navigate({
          pathname: '/(app)/(tabs)/bookings',
          params: { type: BOOKING_TABS.CANCELED },
        })
      }
    } catch (error) {
      if (process.env.NODE_ENV === 'development') {
        console.error('Error cancelling booking:', error)
      }

      toast.show('Cancellation Failed', {
        message: 'There was an error cancelling your booking. Please try again later.',
        type: 'error',
        duration: 3000,
      })
    }

    setAppLoading(false)
  }

  return (
    <Modal
      ref={ref}
      aria-label={`Confirm cancellation of booking with ${doctorName} on ${formatDate(date)}`}
      accessibilityHint="Dialog to confirm or keep this appointment"
      {...props}>
      <YStack alignItems="center" paddingHorizontal={42} gap={32} paddingVertical={32}>
        <YStack
          height={100}
          width={100}
          borderRadius={100}
          backgroundColor="#e0f7f9"
          alignItems="center"
          justifyContent="center">
          <CheckCircle color="$teal" />
        </YStack>

        <Heading size="extraLarge">Confirm Cancellation</Heading>

        <Text size="small" textAlign="center">
          Are you sure you want to cancel your appointment with{' '}
          <Text size="small" fontWeight="700">
            {doctorName}
          </Text>{' '}
          on{' '}
          <Text size="small" fontWeight="700">
            {date}
          </Text>
          ?
        </Text>

        <YStack gap={10} width="100%">
          <Button
            onPress={handleCancelBooking}
            aria-label={`Confirm cancellation of appointment with ${doctorName}`}
            role="button"
            accessibilityHint="Cancels the appointment and navigates to cancelled bookings">
            Yes, Cancel
          </Button>
          <Button
            variant="secondary"
            backgroundColor="transparent"
            color="$teal"
            borderWidth={0}
            onPress={handleClose}
            aria-label={`Keep appointment with ${doctorName}`}
            accessibilityHint="Closes the dialog and keeps the appointment"
            role="button">
            No, Keep Appointment
          </Button>
        </YStack>
      </YStack>
    </Modal>
  )
}

export default CancelBookingModal
