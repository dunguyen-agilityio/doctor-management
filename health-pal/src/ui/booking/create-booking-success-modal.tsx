import { router } from 'expo-router'

import { YStack } from 'tamagui'

import { Button } from '@theme/button'
import { Heading } from '@theme/heading'
import { Text } from '@theme/text'

import { ShieldTick } from '@icons'

import Modal from '@app/components/modal'
import { queryClient } from '@app/react-query.config'
import { BOOKING_TABS } from '@app/types/booking'
import { ModalRef } from '@app/types/modal'

type Props = {
  ref?: React.RefObject<ModalRef | null>
  date: string
  time: string
}

export function CreateBookingSuccessModal({ ref, date, time }: Readonly<Props>) {
  const handleDone = async () => {
    await queryClient.invalidateQueries({
      queryKey: [`bookings-${BOOKING_TABS.UPCOMING}`],
    })
    router.replace('/(app)/(tabs)/bookings')
    ref?.current?.close()
  }

  const handleClose = () => {
    ref?.current?.close()
  }

  return (
    <Modal ref={ref}>
      <YStack alignItems="center" paddingHorizontal={48} gap={32} paddingVertical={32}>
        <YStack
          h={130}
          width={130}
          borderRadius={130}
          backgroundColor="$lightTeal"
          alignItems="center"
          justifyContent="center">
          <ShieldTick />
        </YStack>

        <Heading size="extraLarge" fontWeight="600">
          Congratulations!
        </Heading>

        <Text size="small" color="$gray10" textAlign="center" maxWidth={252}>
          {`Your appointment with Dr. David Patel is confirmed for ${date}, at ${time}.`}
        </Text>
        <YStack width={'100%'} gap={18}>
          <Button onPress={handleDone}>Done</Button>
          <Button
            variant="outlined"
            borderWidth={0}
            color="$grey500"
            fontWeight="400"
            onPress={handleClose}>
            Edit your appointment
          </Button>
        </YStack>
      </YStack>
    </Modal>
  )
}
