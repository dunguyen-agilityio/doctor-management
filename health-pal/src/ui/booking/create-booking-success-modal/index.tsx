import { ShieldTick } from '@/icons'
import { formatShortTime } from '@/utils/date'
import { useFormContext } from 'react-hook-form'

import dayjs from 'dayjs'
import { router } from 'expo-router'

import { YStack } from 'tamagui'

import { ROUTES } from '@/constants'

import { Button, Heading, Text } from '@/components/common'
import Modal from '@/components/common/modal'

import { BOOKING_TABS, BookingForm } from '@/types/booking'
import { ModalRef } from '@/types/modal'

import { queryClient } from '@react-query.config'

type Props = {
  ref?: React.RefObject<ModalRef | null>
}

export const CreateBookingSuccessModal = ({ ref }: Readonly<Props>) => {
  const { watch } = useFormContext<BookingForm>()

  const date = dayjs(watch('date'))
  const formattedDate = date.format('YYYY-MM-DD')
  const formattedTime = formatShortTime(date)

  const handleDone = async () => {
    await queryClient.invalidateQueries({ queryKey: ['bookings', BOOKING_TABS.UPCOMING] })
    router.replace(ROUTES.BOOKINGS)
    ref?.current?.close()
  }

  const handleClose = () => {
    ref?.current?.close()
  }

  return (
    <Modal ref={ref}>
      <YStack alignItems="center" paddingHorizontal={48} gap={32} paddingVertical={32}>
        <YStack
          h={130}
          width={130}
          borderRadius={130}
          backgroundColor="$lightTeal"
          alignItems="center"
          justifyContent="center">
          <ShieldTick />
        </YStack>

        <Heading size="extraLarge" fontWeight="600">
          Congratulations!
        </Heading>

        <Text size="small" color="$gray10" textAlign="center" maxWidth={252}>
          {`Your appointment with Dr. David Patel is confirmed for ${formattedDate}, at ${formattedTime}.`}
        </Text>
        <YStack width={'100%'} gap={18}>
          <Button onPress={handleDone}>Done</Button>
          <Button
            variant="outlined"
            borderWidth={0}
            color="$grey500"
            fontWeight="400"
            onPress={handleClose}>
            Edit your appointment
          </Button>
        </YStack>
      </YStack>
    </Modal>
  )
}
