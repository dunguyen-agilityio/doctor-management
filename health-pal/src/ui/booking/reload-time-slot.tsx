import { Button, Heading, Text, XStack, YStack } from '@theme'

import { Modal } from '@app/components'
import { TBookingCard } from '@app/models/booking'
import { ModalRef } from '@app/types/modal'

interface Props extends Pick<TBookingCard, 'date' | 'time'> {
  ref: React.RefObject<ModalRef | null>
  onReload: () => void
}

const ReloadTimeSlotConfirmModal = ({ ref, date, onReload, time }: Props) => {
  const handleClose = () => {
    ref.current?.close()
  }

  return (
    <Modal ref={ref} contentProps={{ minHeight: 200 }}>
      <YStack padding={16} gap={16} alignItems="center" flex={1}>
        <Heading size="extraLarge">Time Slot Unavailable</Heading>
        <Text size="small">
          The selected time slot ({date} - {time}) is no longer available. Would you like to reload
          the time slots?
        </Text>
        <XStack width="100%" flex={1} alignItems="flex-end" justifyContent="space-around">
          <Button onPress={handleClose} variant="secondary">
            <Text>Cancel</Text>
          </Button>
          <Button onPress={onReload}>Reload Time Slots</Button>
        </XStack>
      </YStack>
    </Modal>
  )
}

export default ReloadTimeSlotConfirmModal
