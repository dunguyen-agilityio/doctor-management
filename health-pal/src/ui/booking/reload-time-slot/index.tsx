import { formatDate } from '@/utils/date'
import { useFormContext } from 'react-hook-form'

import { Button, Heading, Modal, Text, XStack, YStack } from '@/components'

import { BookingForm } from '@/types/booking'
import { ModalRef } from '@/types/modal'

interface Props {
  ref: React.RefObject<ModalRef | null>
  onReload: () => void
}

const ReloadTimeSlotConfirmModal = ({ ref, onReload }: Props) => {
  const { watch } = useFormContext<BookingForm>()

  const date = watch('date')
  const formattedDate = formatDate(date)
  const handleClose = () => {
    ref.current?.close()
  }

  return (
    <Modal ref={ref} minHeight={200}>
      <YStack padding={16} gap={16} alignItems="center" flex={1}>
        <Heading size="extraLarge">Time Slot Unavailable</Heading>
        <Text size="small">
          The selected time slot ({formattedDate}) is no longer available. Would you like to reload
          the time slots?
        </Text>
        <XStack width="100%" flex={1} alignItems="flex-end" justifyContent="space-around">
          <Button onPress={handleClose} variant="secondary">
            <Text>Cancel</Text>
          </Button>
          <Button onPress={onReload}>Reload Time Slots</Button>
        </XStack>
      </YStack>
    </Modal>
  )
}

export default ReloadTimeSlotConfirmModal
