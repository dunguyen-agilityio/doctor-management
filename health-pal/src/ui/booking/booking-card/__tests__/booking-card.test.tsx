import { MOCK_USER } from '@/mocks/user'
import { formatDate } from '@/utils/date'
import { render, screen } from '@utils-test'

import dayjs from 'dayjs'

import { useRequireAuth } from '@/hooks/use-require-auth'

import { BOOKING_TABS } from '@/types/booking'

import BookingCard from '..'

jest.mock('@/hooks/use-require-auth')

describe('BookingCard', () => {
  const defaultProps = {
    date: dayjs('2025-07-01'),
    time: '10:00 AM',
    doctorName: 'Dr. John Doe',
    specialty: 'Cardiology',
    doctorAvatar: 'https://example.com/avatar.jpg',
    address: '123 Main St, City',
    documentId: 'booking1',
    doctorId: 1,
    doctorDocId: 'doc-ref1',
    type: BOOKING_TABS.UPCOMING,
    id: 1,
  }

  beforeEach(() => {
    ;(useRequireAuth as jest.Mock).mockImplementation(() => ({
      session: { user: MOCK_USER },
    }))
  })

  afterEach(() => {
    jest.clearAllMocks()
  })

  it('renders card content', () => {
    render(<BookingCard {...defaultProps} />)

    expect(screen.getByLabelText('Booking with Dr. John Doe')).toBeTruthy()
    expect(screen.getByText(formatDate(defaultProps.date))).toBeTruthy()
    expect(screen.getByText('Dr. John Doe')).toBeTruthy()
    expect(screen.getByText('Cardiology')).toBeTruthy()
    expect(screen.getByText('123 Main St, City')).toBeTruthy()
    expect(screen.getByTestId('cloudinary-image')).toHaveProp(
      'source',
      expect.arrayContaining([
        {
          uri: 'https://example.com/avatar.jpg',
        },
      ]),
    )
    expect(screen.getByTestId('booking-action')).toBeTruthy()
  })

  it('has correct accessibility attributes', () => {
    render(<BookingCard {...defaultProps} />)

    const card = screen.getByLabelText('Booking with Dr. John Doe')
    expect(card).toHaveProp('aria-label', 'Booking with Dr. John Doe')
  })

  it('matches snapshot', () => {
    const { toJSON } = render(<BookingCard {...defaultProps} />)
    expect(toJSON()).toMatchSnapshot()
  })
})
