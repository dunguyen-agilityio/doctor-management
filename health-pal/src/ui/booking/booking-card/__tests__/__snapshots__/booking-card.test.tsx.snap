// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BookingCard matches snapshot 1`] = `
[
  <View
    aria-label="Notifications (F8)"
    pointerEvents="box-none"
    role="region"
    style={
      {
        "bottom": 0,
        "flexDirection": "column",
        "left": 0,
        "maxWidth": "100%",
        "position": "absolute",
        "right": 0,
        "top": 0,
        "zIndex": 100000,
      }
    }
    tabIndex={-1}
  >
    <View
      focusable={false}
      pointerEvents="box-none"
      style={
        {
          "flexDirection": "column",
          "maxWidth": "100%",
          "position": "absolute",
        }
      }
    />
  </View>,
  <View
    aria-label="Booking with Dr. John Doe"
    style={
      {
        "backgroundColor": "#ffffff",
        "borderBottomColor": "#f3f4f6",
        "borderBottomLeftRadius": 12,
        "borderBottomRightRadius": 12,
        "borderBottomWidth": 0.5,
        "borderLeftColor": "#f3f4f6",
        "borderLeftWidth": 0.5,
        "borderRightColor": "#f3f4f6",
        "borderRightWidth": 0.5,
        "borderStyle": "solid",
        "borderTopColor": "#f3f4f6",
        "borderTopLeftRadius": 12,
        "borderTopRightRadius": 12,
        "borderTopWidth": 0.5,
        "flexDirection": "column",
        "marginBottom": 10,
        "paddingBottom": 10,
        "paddingLeft": 10,
        "paddingRight": 10,
        "paddingTop": 10,
        "position": "relative",
        "shadowColor": "rgb(0,0,0)",
        "shadowOffset": {
          "height": 2,
          "width": 0,
        },
        "shadowOpacity": 0.0392156862745098,
        "shadowRadius": 4,
      }
    }
    testID="booking-card"
  >
    <View
      style={
        {
          "backgroundColor": "transparent",
          "flexDirection": "column",
          "marginBottom": "auto",
          "paddingBottom": 8,
          "paddingLeft": 8,
          "paddingRight": 8,
          "paddingTop": 8,
          "zIndex": 10,
        }
      }
    >
      <Text
        style={
          {
            "color": "#1c2a3a",
            "fontFamily": "Inter_700Bold",
            "fontSize": 14,
            "lineHeight": 24,
          }
        }
        suppressHighlighting={true}
      >
        2025-07-01 - 00:00 AM
      </Text>
      <View
        style={[]}
      />
    </View>
    <View
      style={
        {
          "backgroundColor": "transparent",
          "flexDirection": "row",
          "marginBottom": 0,
          "marginTop": "auto",
          "zIndex": 5,
        }
      }
    >
      <View
        style={
          {
            "backgroundColor": "white",
            "flex": 1,
            "flexDirection": "column",
          }
        }
      >
        <View
          style={
            {
              "backgroundColor": "white",
              "flexDirection": "row",
              "gap": 10,
            }
          }
        >
          <View
            style={
              {
                "borderRadius": 8,
                "height": 109,
                "overflow": "hidden",
                "width": 109,
              }
            }
          >
            <ViewManagerAdapter_ExpoImage
              cachePolicy="disk"
              containerViewRef={"[React.ref]"}
              contentFit="cover"
              contentPosition={
                {
                  "left": "50%",
                  "top": "50%",
                }
              }
              height="100%"
              nativeViewRef={"[React.ref]"}
              onError={[Function]}
              onLoad={[Function]}
              onLoadEnd={[Function]}
              onLoadStart={[Function]}
              onProgress={[Function]}
              placeholder={
                [
                  {
                    "height": 16,
                    "uri": "blurhash:/LDPzx.00%5B93GrBXnp0McR5X9NHn%23",
                    "width": 16,
                  },
                ]
              }
              source={
                [
                  {
                    "uri": "https://example.com/avatar.jpg",
                  },
                ]
              }
              style={
                {
                  "height": "100%",
                  "width": "100%",
                }
              }
              testID="cloudinary-image"
              transition={null}
              width="100%"
            />
            <View
              style={
                {
                  "flexDirection": "column",
                  "left": "50%",
                  "position": "absolute",
                  "top": "50%",
                  "transform": [
                    {
                      "translateX": -12,
                    },
                    {
                      "translateY": -12,
                    },
                  ],
                }
              }
            >
              <ActivityIndicator
                color="#1c64f2"
                size="small"
              />
            </View>
          </View>
          <View
            style={
              {
                "backgroundColor": "white",
                "flexDirection": "column",
                "paddingBottom": 14,
                "paddingTop": 14,
              }
            }
          >
            <Text
              style={
                {
                  "color": "#1c2a3a",
                  "fontFamily": "System",
                  "fontSize": 16,
                  "fontWeight": "bold",
                  "lineHeight": 24,
                }
              }
              suppressHighlighting={true}
            >
              Dr. John Doe
            </Text>
            <Text
              style={
                {
                  "color": "#6b7280",
                  "fontFamily": "Inter_400Regular",
                  "fontSize": 14,
                  "lineHeight": 24,
                }
              }
              suppressHighlighting={true}
            >
              Cardiology
            </Text>
            <Text
              numberOfLines={1}
              style={
                {
                  "color": "#9ca3af",
                  "fontFamily": "Inter_400Regular",
                  "fontSize": 12,
                  "lineHeight": 24,
                  "maxWidth": 250,
                }
              }
              suppressHighlighting={true}
            >
              123 Main St, City
            </Text>
          </View>
        </View>
        <View
          style={[]}
        />
        <View
          style={
            {
              "flexDirection": "row",
              "gap": 10,
            }
          }
          testID="booking-action"
        >
          <View
            accessibilityHint="Opens a confirmation dialog to cancel this appointment"
            aria-label="Cancel booking with Dr. John Doe on 2025-07-01 - 00:00 AM"
            focusVisibleStyle={{}}
            focusable={true}
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            role="button"
            style={
              {
                "alignItems": "center",
                "backgroundColor": "#e5e7eb",
                "borderBottomColor": "transparent",
                "borderBottomLeftRadius": 60,
                "borderBottomRightRadius": 60,
                "borderBottomWidth": 1,
                "borderLeftColor": "transparent",
                "borderLeftWidth": 1,
                "borderRightColor": "transparent",
                "borderRightWidth": 1,
                "borderStyle": "solid",
                "borderTopColor": "transparent",
                "borderTopLeftRadius": 60,
                "borderTopRightRadius": 60,
                "borderTopWidth": 1,
                "color": "#1c2a3a",
                "flex": 1,
                "flexDirection": "row",
                "flexWrap": "nowrap",
                "fontWeight": "500",
                "height": 37,
                "justifyContent": "center",
                "opacity": 1,
                "paddingLeft": 8,
                "paddingRight": 8,
              }
            }
          >
            <Text
              lineBreakMode="clip"
              numberOfLines={1}
              style={
                {
                  "color": "#1c2a3a",
                  "flexGrow": 0,
                  "flexShrink": 1,
                  "fontFamily": "Inter_500Medium",
                  "fontSize": 14,
                  "lineHeight": 21,
                  "userSelect": "none",
                }
              }
              suppressHighlighting={true}
            >
              Cancel
            </Text>
          </View>
          <View
            accessibilityHint="Navigates to the booking screen to change the date or time of this appointment"
            aria-label="Reschedule booking with Dr. John Doe"
            focusVisibleStyle={{}}
            focusable={true}
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            role="button"
            style={
              {
                "alignItems": "center",
                "backgroundColor": "#1c2a3a",
                "borderBottomColor": "transparent",
                "borderBottomLeftRadius": 60,
                "borderBottomRightRadius": 60,
                "borderBottomWidth": 1,
                "borderLeftColor": "transparent",
                "borderLeftWidth": 1,
                "borderRightColor": "transparent",
                "borderRightWidth": 1,
                "borderStyle": "solid",
                "borderTopColor": "transparent",
                "borderTopLeftRadius": 60,
                "borderTopRightRadius": 60,
                "borderTopWidth": 1,
                "color": "#ffffff",
                "flex": 1,
                "flexDirection": "row",
                "flexWrap": "nowrap",
                "fontWeight": "500",
                "height": 37,
                "justifyContent": "center",
                "opacity": 1,
                "paddingLeft": 8,
                "paddingRight": 8,
              }
            }
          >
            <Text
              lineBreakMode="clip"
              numberOfLines={1}
              style={
                {
                  "color": "#ffffff",
                  "flexGrow": 0,
                  "flexShrink": 1,
                  "fontFamily": "Inter_500Medium",
                  "fontSize": 14,
                  "lineHeight": 21,
                  "userSelect": "none",
                }
              }
              suppressHighlighting={true}
            >
              Reschedule
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>,
]
`;
